### PR TITLE
Fix broken auth in swagger

### PR DESCRIPTION
**Pull request by guillaumesasdy**
Hello @ttu 

Here is the fix for all authentification methods: `AllowAll` doesn't change.
For `basic` and `token`, there is the Authenticate button I showed as screenshot in #71 

I followed your advice to keep the `logout` endpoint for the `token` method only. I deleted the authenticate endpoint, which would be duplicated.

Concerning the code:  I removed as much as I could the code from `Startup.cs` class to move it in classes in Authentication folder. This implies some refactoring, I let you tell me if you are OK with that. All the logic should work as before.

### CODE REVIEW

File: FakeServer/Authentication/JWT/TokenProviderMiddleware.cs [hunk 1]
```diff
@@ -92,7 +92,7 @@ private async Task<(string username, string password, bool validData)> GetFromJs
                     var authHeader = context.Request.Headers["Authorization"].ToString();
                     var authString = authHeader.Substring("Basic ".Length).Trim();
                     var credentials = Encoding.UTF8.GetString(Convert.FromBase64String(authString)).Split(':');
-                    return (credentials[1], credentials[0], true);
+                    return (credentials[0], credentials[1], true);
```
Comment (ttu) [inline FakeServer/Authentication/JWT/TokenProviderMiddleware.cs:95]: Good catch 👍

File: FakeServer/Authentication/ConfigureAuthentication.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                return configuration["Authentication:AuthenticationType"] == "token"
+                    ? AuthenticationType.JwtBearer
+                    : AuthenticationType.Basic;
+            }
+            else return AuthenticationType.AllowAll;
+        }
+
+        public static IServiceCollection AddAuthentication(this IServiceCollection services, AuthenticationType type)
+        {
+            switch (type)
```
Comment (ttu) [inline FakeServer/Authentication/ConfigureAuthentication.cs:27]: Use switch expression

File: FakeServer/Authentication/ConfigureAuthentication.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        public static AuthenticationType ReadType(IConfiguration configuration)
+        {
+            if (configuration.GetValue<bool>("Authentication:Enabled"))
+            {
+                return configuration["Authentication:AuthenticationType"] == "token"
+                    ? AuthenticationType.JwtBearer
+                    : AuthenticationType.Basic;
+            }
+            else return AuthenticationType.AllowAll;
```
Comment (ttu) [inline FakeServer/Authentication/ConfigureAuthentication.cs:22]: else not needed

File: FakeServer/Startup.cs [hunk 4]
```diff
@@ -106,23 +89,6 @@ public void ConfigureServices(IServiceCollection services)
                 jsonFormatter.SupportedMediaTypes.Add(new MediaTypeHeaderValue(Constants.JsonMergePatch));
                 jsonFormatter.SupportedMediaTypes.Add(new MediaTypeHeaderValue(Constants.MergePatchJson));
             });
-
-            services.AddSwaggerGen(c =>
-            {
-                c.SwaggerDoc("v1", new OpenApiInfo { Title = "Fake JSON API", Version = "v1" });
-
-                var basePath = PlatformServices.Default.Application.ApplicationBasePath;
```
Comment (ttu) [inline FakeServer/Startup.cs:114]: Is this still required so we get all comments to SwaggerUI?

**Comment by odin88** [comment]
Any updates on this ?

**Comment by ttu** [comment]
No update on this. This is a working solution with the changes I suggested. I haven’t found time to move it to the current main branch.

@odin88 interested to finish this?

**Comment by Lucarine** [comment]
@ttu This doesn't look like it's been worked on in a while. Is it OK if I finish up the rest of the work needed on this PR?

**Comment by ttu** [comment]
@Lucarine Hi! It seems that no one is working on this, so it is great if you can finish this. We have changed the framework to .NET 5, so it might be better that you create a new branch from master and move this changes there.

**Comment by ttu** [comment]
Closed as we have fix in other PR

**Pull request closed without merging**